odl/tomo/operators/spect_trafo.py

```diff
@@ -0,0 +1,254 @@
+    def __init__(self, dom, geometry, attenuation=None, psf=None,
+                 impl='niftyrec'):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        dom : `DiscreteLp`
+            Discretized space, a 3 dimensional domain of the forward projector
+        geometry : geometry from :class:`~odl.tomo.geometry.spect`
+            The geometry of the transform. An ODL geometry instance
+            that contains information about the operator domain
+        attenuation : `element-like` structure of the shape
+            `dom.shape`, optional
+            Linear attenuation map for SPECT.
+        psf : `element-like` structure, optional
+            Point Spread Functions for the correction of the
+            collimator-detector response
+        impl={'niftyrec', 'niftyrec_gpu'}, optional
+            Implementation back-end for the transform. Supported back-ends:
+
+            'niftyrec' : `NiftyRec <http://niftyrec.scienceontheweb.net/\
+wordpress/>`_ tomography toolbox using the `niftypy <https://github.com/\
+spedemon/niftypy>`_ Python wrapper and CPU
+
+            'niftyrec_gpu' : `NiftyRec <http://niftyrec.scienceontheweb.net/\
+wordpress/>`_ tomography toolbox using the `NiftyPy <https://github.com/\
+spedemon/niftypy>`_ Python wrapper and GPU
+        """
+        impl, impl_in = str(impl).lower(), impl
+        if impl not in _SUPPORTED_IMPL:
+            raise ValueError('implementation {!r} not supported.'
+                             ''.format(impl_in))
+
+        if impl.startswith('niftyrec'):
+            if not NIFTYREC_AVAILABLE:
+                raise ValueError('NiftyRec back-end not available.')
+            if impl == 'niftyrec_gpu' and dom.shape != (128, 128, 128):
+                raise NotImplementedError('NiftyRec GPU back-end only '
+                                          'supports volume size '
+                                          '(128, 128, 128), got {}'
+                                          ''.format(dom.shape))
+            if impl == 'niftyrec_gpu':
+                self._use_gpu = True
+            elif impl == 'niftyrec':
+                self._use_gpu = False
+
+        self._geometry = geometry
+        det_nx, det_ny = geometry.det_grid.shape
+        startAngle = geometry.angles[0]
+        stopAngle = geometry.angles[-1]
+        N_proj = geometry.motion_grid.size
+
+        self._cameras = np.linspace(startAngle, stopAngle, N_proj,
+                                    dtype=np.float32).reshape((N_proj, 1))
+        if psf is not None:
+            self._psf = np.asarray(psf)
+        else:
+            self._psf = psf
+        if attenuation is not None:
+            self._attenuation = self.domain.element(attenuation)
+        else:
+            self._attenuation = attenuation
+        self._impl = impl
+
+        ran = uniform_discr_frompartition(geometry.partition)
+        super().__init__(dom, ran, linear=True)
+
+    def _call(self, x):
```

Comment:
Currently, the Nifty back-end is hard-coded. To make this more extendible, we should rename the current `_call` to `_call_nifty` and write a new `_call` which just decides which one of the `_call_*` to take. See [here](https://github.com/odlgroup/odl/blob/master/odl/trafos/fourier.py#L1592-L1617) for an example. The main documentation goes into the `_call` method, the other guys don't need so much documentation. Check [here](https://github.com/odlgroup/odl/blob/master/odl/trafos/fourier.py#L1660-L1672) for an example.

If we get more back-ends, we will have to factor out those specific parts into own files like for the ray transform, but for now it's okay like this.

Note: This requires us to store `impl`. Would be worth a property also.